zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -505,7 +511,14 @@ where
                 //
                 // Optional TODO:
                 // - add a `MempoolChange` type with an `async changed()` method (like `ChainTip`)
-                let mempool_txs = fetch_mempool_transactions(mempool.clone()).await?;
+                let Some(mempool_txs) =
+                    fetch_mempool_transactions(mempool.clone(), chain_tip_and_local_time.tip_hash)
+                        .await?
+                        // Omit mempool transactions from the template if called without a long poll id, or
+                        // continue to the next iteration of the loop to make fresh state and mempool requests.
```

Comment:
Can we add the reason we're doing this here?
(It's ok to skip implementation details that are in the code!)

```suggestion
                        // If the mempool and state responses are out of sync:
                        // - if we are not long polling, omit mempool transactions from the template,
                        // - if we are long polling, continue to the next iteration of the loop to make fresh state and mempool requests.
```